src/loaders.rs

```diff
@@ -31,6 +32,45 @@ fn safe_join(directory: &Path, template_name: &str) -> Option<PathBuf> {
     }
 }
 
+#[cached(
+    size = 128,          // Cache size
+    result = true,       // Cache Result type
+    key = "String",      // Use owned String as key
+    convert = r##"{ dirname.to_string() }"## // Convert &str to String
+)]
```

Comment:
What is the goal of this caching? I wonder if this is premature optimisation.

Reply:
Ah, I see that [Django includes an `lru_cache`](https://github.com/django/django/blob/ea1e3703bee28bfbe4f32ceb39ad31763353b143/django/template/utils.py#L97) in its implementation. This probably is worth caching then.